src/classlibnative/bcltype/bignum.cpp

```diff
@@ -0,0 +1,599 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: bignum.cpp
+//
+
+//
+
+#include "bignum.h"
+#include <intrin.h>
+
+constexpr UINT32 BigNum::m_power10UInt32Table[UINT32POWER10NUM];
+BigNum BigNum::m_power10BigNumTable[BIGPOWER10NUM];
+BigNum::StaticInitializer BigNum::m_initializer;
+
+BigNum::BigNum()
+    :m_len(0) // Note: we do not zeroing m_blocks due to performance.
+{
+}
+
+BigNum::BigNum(UINT32 value)
+{
+    SetUInt32(value);
+}
+
+BigNum::BigNum(UINT64 value)
+{
+    SetUInt64(value);
+}
+
+BigNum::~BigNum()
+{
+}
+
+BigNum& BigNum::operator=(const BigNum &rhs)
+{
+    memcpy(m_blocks, rhs.m_blocks, ((UINT32)rhs.m_len) * sizeof(UINT32));
+    m_len = rhs.m_len;
+
+    return *this;
+}
+
+int BigNum::Compare(const BigNum& lhs, const BigNum& rhs)
+{
+    int lenDiff = (int)lhs.m_len - (int)rhs.m_len;
```

Comment:
Could you leave a comment indicating that this is never expected to over/underflow due to the min/max of `m_len`?

Having an assert that `m_len` is never less than 0 or greater than `max length` would be good too